victor/src/layout/mod.rs

```diff
@@ -19,43 +20,82 @@ impl crate::dom::Document {
         layout_document(&box_tree, viewport)
     }
 }
+
 fn layout_document(
     box_tree: &BoxTreeRoot,
     viewport: crate::primitives::Size<crate::primitives::CssPx>,
 ) -> Vec<Fragment> {
+    let inline_size = Length { px: viewport.width };
+
     // FIXME: use the document’s mode:
     // https://drafts.csswg.org/css-writing-modes/#principal-flow
     let initial_containing_block = ContainingBlock {
-        inline_size: Length { px: viewport.width },
+        inline_start_from_absolute_containing_block: Length::zero(),
+        inline_size,
         block_size: Some(Length {
             px: viewport.height,
         }),
         mode: (WritingMode::HorizontalTb, Direction::Ltr),
     };
 
-    let (fragments, _) = box_tree.layout(&initial_containing_block);
+    let (mut fragments, absolutely_positioned_fragments, _) =
+        box_tree.layout(&initial_containing_block, &initial_containing_block, 0);
+    fragments.extend(
+        absolutely_positioned_fragments
+            .into_iter()
+            .map(|fragment| Fragment::Box(fragment.contents)),
+    );
     fragments
 }
 
+struct AbsoluteContext<'a> {
+    containing_block: &'a ContainingBlock,
+    absolutely_positioned_fragments: Mutex<Vec<AbsolutelyPositionedFragment>>,
+}
+
+struct AbsolutelyPositionedFragment {
+    index: usize,
+    uses_static_block_position: bool,
+    contents: BoxFragment,
+}
+
 struct ContainingBlock {
+    inline_start_from_absolute_containing_block: Length,
```

Comment:
Tracking this information this way won’t work when the writing mode can change between horizontal or vertical between an abspos box and its containing block. In a follow-up PR, could you replace this with `uses_static_inline_position: bool` in `AbsolutelyPositionedFragment`?

Reply:
Ack, didn't know that. I'll probably try to support actual different writing modes then because I don't want to put the field and do nothing with it.

Reply:
Hmm I really think you don’t want to be implementing all of https://drafts.csswg.org/css-writing-modes/#orthogonal-flows right now.

Reply:
This is gone too.